src/person.cpp

```diff
@@ -1617,6 +1664,73 @@ AnimateResult Guest::VisitRideOnAnimate(RideInstance *ri, TileEdge exit_edge)
 	return OAR_CONTINUE;
 }
 
+/** Pixel positions of guests sitting on a bench. */
+static XYZPoint16 _bench_pix_pos[4 /* TileEdge */][2 /* Left = 0, Right = 1 */] = {
+	{XYZPoint16(  0, 160, 0), XYZPoint16(  0,  96, 0)},
+	{XYZPoint16(160, 255, 0), XYZPoint16( 96, 255, 0)},
+	{XYZPoint16(255,  96, 0), XYZPoint16(255, 160, 0)},
+	{XYZPoint16( 96,   0, 0), XYZPoint16(160,   0, 0)},
+};
+
+AnimateResult Guest::InteractWithPathObject(PathObjectInstance *obj)
+{
+	const TileEdge edge = this->GetCurrentEdge();
+	if (obj->type == &PathObjectType::LITTERBIN && this->has_wrapper && obj->GetFreeBinCapacity(edge) > 0) {
+		/* Throw litter in the bin, then keep walking */
+		obj->AddItemToBin(edge);
+		this->has_wrapper = false;
+	} else if (obj->type == &PathObjectType::BENCH && this->nausea > 40 &&
+			(obj->GetLeftGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH ||
+			obj->GetRightGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH)) {
+		/* Sit down and remain there for a while. */
+		if (obj->GetRightGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH) {
+			obj->SetRightGuest(edge, this->id);
+			this->pix_pos = _bench_pix_pos[edge][1];
+		} else {
+			obj->SetLeftGuest(edge, this->id);
+			this->pix_pos = _bench_pix_pos[edge][0];
+		}
+		this->activity = GA_RESTING;
+		this->StartAnimation(_guest_bench[edge]);
+		return OAR_OK;
+	} else if (this->happiness < 40) {
+		/* Smash something up, then keep walking. */
+		for (int8 dx = -2; dx <= 2; dx++) {
+			for (int8 dy = -2; dy <= 2; dy++) {
+				if (!IsVoxelstackInsideWorld(this->vox_pos.x + dx, this->vox_pos.y + dy)) continue;
+				for (int8 dz = -1; dz <= 1; dz++) {
+					const Voxel *vx = _world.GetVoxel(XYZPoint16(this->vox_pos.x + dx, this->vox_pos.y + dy, this->vox_pos.z + dz));
+					if (vx == nullptr) continue;
+					for (VoxelObject *o = vx->voxel_objects; o != nullptr; o = o->next_object) {
+						if (dynamic_cast<Guard*>(o) != nullptr) {
+							/* A security guard is nearby, so no vandalism just yet. */
+							return OAR_CONTINUE;
+						}
+					}
+				}
+			}
+		}
+
+		obj->Demolish(edge);
```

Comment:
You ever saw this happen in-game in the original? I never did :)

Reply:
I was never able to actually *see* a guest doing it, so I have no idea if there's an animation for it (I didn't add one). But my benches are forever getting smashed up and then the vandals complain about the extent of vandalism. When I have achieved all objectives and have to spend the remaining year or so sitting around waiting, my main occupation is to go round the park and replace damaged park property.

---

Since we're on the topic, what would you think about enabling mechanics to repair broken path objects? Would be much more convenient than having to seek and replace them all by hand…

Reply:
I never had much trouble with vandalism, only at times when I couldn't buy new rides etc to entertain everybody, or enough handyman to keep the park nice and tidy. I also had low costs for using a ride, so lots of happy customers, I guess :)
Likely my profit was much lower than it needed to be, but who cares. At some point money just pours in by itself.

As for keeping handyman busy, perhaps add a cheats corner, or "enhanced gameplay", or so? Eventually we'll need it anyway :)